src/angularjs-toast.coffee

```diff
@@ -87,6 +95,8 @@ angular.module 'angularjsToast', ['ngSanitize', 'ngAnimate']
       # push elements to array
       pushToArray = ->
         if args.insertFromTop then scope.$toastMessages.unshift(json) else scope.$toastMessages.push(json)
+        #scope update bug correction
+        _scopeUpdate()          
```

Comment:
Manually Updating scope inside a library is not recommended.

Reply:
so i should do it outside the library?

Reply:
> @LeonadoRivaldo I have requested few changes. make the changes I will merge this. Follow Conventional https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits. While you make a commit.

I'll make the changes today on my lunch time, and post asap!

Reply:
> so i should do it outside the library?

Yes. Because it is too costly and might be destructive to use `apply` always and unnecessarily. It is safe to use outside of a library.

In your case You can store it to an intermediate variable, use `angular's built in watch function` and if you find changes. Take necessary action.